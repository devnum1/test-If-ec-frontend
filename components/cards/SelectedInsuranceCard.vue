// 01EZTX9M3AWK0ZB16B6KYR56F1

<template>
  <div class="selected-insurance-card">
    <div class="insurance-head">
      <div class="insurance-head__inner">
        {{ compensation }}{{ $t('insurance.hoshou_plan') }}
      </div>
    </div>
    <div class="insurance-body">
      <div class="insurance-body__inner">
        <div>{{ $t('common.total') }}:&nbsp;</div>
        <div class="insurance-body__inner-price">
          {{ price | formatPrice }}
        </div>
        <div>&nbsp;{{ $t('common.yen_per_month') }}</div>
      </div>
    </div>
  </div>
</template>
<script type="text/javascript">
export default {
  props: {
    compensation: {
      type: String,
      default: '',
    },
    price: {
      type: [String, Number],
      default: '',
    },
  },
  data() {
    return {
      progress: 1,
      formatedPrice: '',
    };
  },
};
</script>

<style lang="scss" scoped>
.selected-insurance-card {
  border: 1px solid rgba($green-border, 0.15);
  border-radius: 6px;
  text-align: center;
  padding-left: 16px;
  padding-right: 16px;
  background-color: white;
  font-family: $font-family-base;

  .insurance-head {
    &__inner {
      padding-top: 22px;
      padding-bottom: 21px;
      border-bottom: 1px dashed rgba($green-border, 0.15);
      font-size: 24px;
      line-height: 30px;
      color: $green;
      font-family: $font-family-base;
    }
  }

  .insurance-body {
    &__inner {
      font-size: 14px;
      padding-top: 11px;
      padding-bottom: 20px;
      font-family: $font-family-base;
      color: $base-gray;

      div {
        display: inline-block;
      }

      &-price {
        font-size: 24px;
        color: $base-black;
        line-height: 30px;
      }
    }
  }
}
</style>
