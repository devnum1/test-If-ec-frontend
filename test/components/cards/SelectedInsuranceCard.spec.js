import { shallowMount, config } from '@vue/test-utils';
import SelectedInsuranceCard from '@/components/cards/SelectedInsuranceCard.vue';
import Vue from 'vue';
import { promisify } from 'util';

config.mocks.$t = (key) => key;

describe('SelectedInsuranceCard', () => {
  const compensation = 52;
  const price = 3725600;
  const sleep = promisify(setTimeout);

  const flushPromises = () => new Promise(setImmediate);
  let wrapper;

  const factory = (computed = {}) => {
    return shallowMount(SelectedInsuranceCard, {
      propsData: {
        compensation: compensation.toString(),
        price: price.toString(),
      },
      mocks: {
        $t: () => {}, // erase all $(t) to test (ignore i18n)
      },
    });
  };

  beforeEach(() => {
    Vue.filter('formatPrice', function formatPrice(value) {
      let val = (value / 1).toFixed(0).replace('.', ',');

      return val.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
    });
    wrapper = factory();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy();
  });

  test('match Snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Card to display right compensation', async () => {
    await flushPromises();

    expect(wrapper.find('.insurance-head__inner').exists()).toBe(true);
    const compensationText = wrapper.find('.insurance-head__inner');
    expect(compensationText.text()).toBe(compensation + '');
  });
});
