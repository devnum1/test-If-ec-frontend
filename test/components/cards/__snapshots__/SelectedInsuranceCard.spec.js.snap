// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedInsuranceCard match Snapshot 1`] = `
"<div class=\\"selected-insurance-card\\">
  <div class=\\"insurance-head\\">
    <div class=\\"insurance-head__inner\\">
      52
    </div>
  </div>
  <div class=\\"insurance-body\\">
    <div class=\\"insurance-body__inner\\">
      <div>
        :&nbsp;
      </div>
      <div class=\\"insurance-body__inner-price\\">
        3,725,600
      </div>
      <div>
        &nbsp;
      </div>
    </div>
  </div>
</div>"
`;
